src/java.desktop/share/classes/java/awt/TextArea.java

```diff
@@ -606,6 +607,15 @@ public Dimension minimumSize() {
         }
     }
 
+    @Override
+    public synchronized void setEditable(boolean b) {
+        super.setEditable(b);
+        Color defaultBackground = UIManager.getColor("TextArea.background");
+        if (!backgroundSetByClientCode) {
+            setBackground(b ? defaultBackground : SystemColor.control, false);
```

Comment:
Is there a reason behind choosing `SystemColor.control` as the default TextComponent system color?
I see another option `SystemColor.text` - the description sounds more close to the background color of text components.

Reply:
the SystemColor.control comes from the docs on TextComponent.setEditable
```
     * If the flag is set to {@code true}, this text component
     * becomes user editable. If the flag is set to {@code false},
     * the user cannot change the text of this text component.
     * By default, non-editable text components have a background color
     * of SystemColor.control.  This default can be overridden by
     * calling setBackground.